tests/plugins/test_add_labels.py

```diff
@@ -704,3 +706,125 @@ def test_add_labels_base_image(tmpdir, docker_tasker, parent, should_fail,
+    (None,  'A',   'A',   'B',   'B',   'B',   'as an alias for label'),
+    (None,  'A',   'B',   'A',   'A',   'A',   'as an alias for label'),
+    (None,  'A',   'B',   'B',   'B',   'B',   'already exists'),
+    (None,  'A',   'B',   'C',   'C',   'C',   'as an alias for label'),
+    ('A',   None,  None,  None,  None,  'A',   'as an alias for label'),
+    ('A',   None,  None,  'A',   None,  'A',   'already exists'),
+    ('A',   None,  None,  'B',   'B',   'B',   'as an alias for label'),
+    ('A',   None,  'A',   None,  'A',   'A',   'as an alias for label'),
+    ('A',   None,  'B',   None,  'B',   'B',   'as an alias for label'),
+    ('A',   None,  'A',   'A',   'A',   'A',   'already exists'),
+    ('A',   None,  'A',   'B',   'B',   'B',   'as an alias for label'),
+    ('A',   None,  'B',   'A',   'A',   'A',   'as an alias for label'),
+    ('A',   None,  'B',   'B',   'B',   'B',   'already exists'),
+    ('A',   None,  'B',   'C',   'C',   'C',   'as an alias for label'),
+    ('A',   'A',   None,  None,  None,  None,  'already exists'),
+    ('A',   'A',   None,  'A',   None,  'A',   'already exists'),
+    ('A',   'A',   None,  'B',   'B',   'B',   'as an alias for label'),
+    ('A',   'A',   'A',   None,  'A',   None,  'already exists'),
+    ('A',   'A',   'B',   None,  'B',   'B',   'as an alias for label'),
+    ('A',   'A',   'A',   'A',   'A',   'A',   'already exists'),
+    ('A',   'A',   'A',   'B',   'B',   'B',   'as an alias for label'),
+    ('A',   'A',   'B',   'A',   'A',   'A',   'as an alias for label'),
+    ('A',   'A',   'B',   'B',   'B',   'B',   'already exists'),
+    ('A',   'A',   'B',   'C',   'C',   'C',   'as an alias for label'),
+    ('A',   'B',   None,  None,  'B',   None,  'as an alias for label'),
+    ('A',   'B',   None,  'A',   None,  'A',   'already exists'),
+    ('A',   'B',   None,  'B',   'B',   'B',   'as an alias for label'),
+    ('A',   'B',   None,  'C',   'C',   'C',   'as an alias for label'),
+    ('A',   'B',   'A',   None,  'A',   'A',   'as an alias for label'),
+    ('A',   'B',   'B',   None,  'B',   None,  'already exists'),
+    ('A',   'B',   'C',   None,  'C',   'C',   'as an alias for label'),
+    ('A',   'B',   'A',   'A',   'A',   'A',   'already exists'),
+    ('A',   'B',   'A',   'B',   'B',   'B',   'as an alias for label'),
+    ('A',   'B',   'A',   'C',   'C',   'C',   'as an alias for label'),
+    ('A',   'B',   'B',   'A',   'A',   'A',   'as an alias for label'),
+    ('A',   'B',   'B',   'B',   'B',   'B',   'already exists'),
+    ('A',   'B',   'B',   'C',   'C',   'C',   'as an alias for label'),
+    ('A',   'B',   'C',   'A',   'A',   'A',   'as an alias for label'),
+    ('A',   'B',   'C',   'B',   'B',   'B',   'as an alias for label'),
+    ('A',   'B',   'C',   'C',   'C',   'C',   'already exists'),
+    ('A',   'B',   'C',   'D',   'D',   'D',   'as an alias for label'),
+])
+def test_release_label(tmpdir, docker_tasker, caplog,
```

Comment:
I don't think this test should have old vs new style labels. It should only be concerned with `'release'` value from base image and `'release'` value from the Dockerfile of the layered image.